distill/core/log.py

```diff
@@ -46,7 +47,8 @@ def __init__(self, data: Union[str, JsonDict], schema=UserAleSchema):
             raise TypeError("ERROR: " + str(type(data)) + " data should be either a string or a JsonDict")
         self.data = schema(**data)
 
-        # TODO: need to create ID field here on object initialization
+        self.id = PKSUID("log", schema._timestamp(self.data))
```

Comment:
Bake the userId into the PKSUID as Evan mentioned in the call to make the log truly globally unique. Ex: log_userId_hash